File: src/main/resources/static/js/applications/domains/park/parkPoi.js
class ParkPoi {
    #id;
    #name;
    #buildingId;
    #floorNo;
    #floorNm;
    #type;
    #property;
    #latitude;
    #longitude;

    constructor(
        id,
        name,
        buildingId,
        floorNo,
        floorNm,
        type,
        property,
        position,
        rotation,
        scale,
        latitude,
        longitude
    ) {
        this.#id = id;
        this.#name = name;
        this.#buildingId = buildingId;
        this.#floorNo = floorNo;
        this.#floorNm = floorNm;
        this.#type = type;
        this.position = position;
        this.rotation = rotation;
        this.scale = scale;
        this.#latitude = latitude;
        this.#longitude = longitude;
        this.#property = property;
    }

    get id() {
        return this.#id;
    }
    get name() {
        return this.#name;
    }
    get buildingId() {
        return this.#buildingId;
    }
    get floorNo() {
        return this.#floorNo;
    }
    get floorNm() {
        return this.#floorNm;
    }
    get type() {
        return this.#type;
    }
    get property() {
        return this.#property;
    }
    get latitude() {
        return this.#latitude;
    }
    get longitude() {
        return this.#longitude;
    }
    // Px.Poi.Add 용
    get poiOptions() {
        return {
            id: this.id,
            displayText: this.name,
            lineHeight: SystemSettingManager.find().poiLineLength ?? 40,
            position: this.position ? this.position : {x:0,y:0,z:0},
            property: this.property,
            rotation: {
                "x": 0,
                "y": 0,
                "z": 0
            },
        };
    }

    removeOn3D(onComplete) {
        Px.Poi.Remove(this.id, onComplete);
    }

    setName(name) {
        this.#name = name;
    }
}

File: src/main/java/com/pluxity/ktds/domains/parking/entity/ParkPoi.java
package com.pluxity.ktds.domains.parking.entity;

import com.pluxity.ktds.domains.building.entity.Building;
import com.pluxity.ktds.domains.building.entity.Spatial;
import com.pluxity.ktds.domains.kiosk.entity.KioskCategory;
import com.pluxity.ktds.domains.parking.dto.ParkPoiResponseDTO;
import com.pluxity.ktds.domains.parking.enums.ParkPoiType;
import com.pluxity.ktds.domains.parking.enums.ParkSideType;
import com.pluxity.ktds.domains.plx_file.entity.FileInfo;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "park_poi")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ParkPoi {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "building_id")
    private Building building;

    @Column
    private Integer floorNo;

    @Column
    private String floorNm;

    @Embedded
    @AttributeOverride(name = "x", column = @Column(name = "position_x"))
    @AttributeOverride(name = "y", column = @Column(name = "position_y"))
    @AttributeOverride(name = "z", column = @Column(name = "position_z"))
    private Spatial position;

    @Embedded
    @AttributeOverride(name = "x", column = @Column(name = "rotation_x"))
    @AttributeOverride(name = "y", column = @Column(name = "rotation_y"))
    @AttributeOverride(name = "z", column = @Column(name = "rotation_z"))
    private Spatial rotation;

    @Embedded
    @AttributeOverride(name = "x", column = @Column(name = "scale_x"))
    @AttributeOverride(name = "y", column = @Column(name = "scale_y"))
    @AttributeOverride(name = "z", column = @Column(name = "scale_z"))
    private Spatial scale;

    @Column(name = "latitude")
    private Long latitude;

    @Column(name = "longitude")
    private Long longitude;

    @Column(name = "park_poi_name")
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "park_poi_type")
    private ParkPoiType poiType;

    @Enumerated(EnumType.STRING)
    @Column(name = "park_side_type")
    private ParkSideType parkSideType;

    public ParkPoiResponseDTO toResponseDTO() {
        return ParkPoiResponseDTO.builder()
                .id(this.id)
                .buildingId(this.building.getId())
                .floorNo(this.floorNo)
                .floorNm(this.floorNm)
                .name(this.name)
                .position(this.position)
                .rotation(this.rotation)
                .scale(this.scale)
                .latitude(this.latitude)
                .longitude(this.longitude)
                .poiType(this.poiType)
                .parkSideType(this.parkSideType)
                .build();

    }
}
